yt_dlp/downloader/fragment.py

```diff
@@ -355,7 +355,8 @@ def decrypt_fragment(fragment, frag_content):
             # not what it decrypts to.
             if self.params.get('test', False):
                 return frag_content
-            return aes_cbc_decrypt_bytes(frag_content, decrypt_info['KEY'], iv)
+            decrypted_data = aes_cbc_decrypt_bytes(frag_content, decrypt_info['KEY'], iv)
+            return decrypted_data[:-decrypted_data[-1]]
```

Comment:
Why?

Deferring to a library implementation, as @shirt-dev originally did, might have detected padding errors, which could indicate corrupt downloads, broken invariants and improper use of cryptography elsewhere. A cutesy one-liner fails to do that. Compare, again, how a by-the-book WebVTT parser allowed us to catch #637. (Yes, I am going to hammer that example again and again.)

I am less convinced that the un-padding should be performed inside `aes_cbc_decrypt_bytes`, but that’s another issue.

Reply:
this code may be called even in the absence of cryptodome. So if we go by the original implementation, a fallback function has to be defined in case the library is absent. So at the time, I took the shortcut of using the "cutesy one-liner" instead. I do agree that a more robust implementation could be useful here

Reply:
If I recall correctly, pycryptodome uses similar code to this for their unpadding function

Reply:
<https://pycryptodome.readthedocs.io/en/latest/src/util/util.html#Crypto.Util.Padding.unpad>:

> **Raises:** `ValueError` – if the padding is incorrect.

The above doesn’t. ([This description agrees with PyCryptodome’s source code](https://github.com/Legrandin/pycryptodome/blob/d8edf1a6a70d3a65dcda18ee24d96161525f2825/lib/Crypto/Util/Padding.py#L85-L106).)

Reply:
I'd rather not abort downloads upon a padding error. If you think we should, please reach out to us on Discord so we can discuss this. https://discord.gg/H5MNcFW63r